src/main/java/net/sf/jabref/gui/desktop/os/NativeDesktop.java

```diff
@@ -1,22 +1,71 @@
 package net.sf.jabref.gui.desktop.os;
 
 import java.io.IOException;
+import java.util.Optional;
 
-public interface NativeDesktop {
-    void openFile(String filePath, String fileType) throws IOException;
+import net.sf.jabref.external.ExternalFileType;
+import net.sf.jabref.external.ExternalFileTypes;
+
+public abstract class NativeDesktop {
```

Comment:
In that case, the concrete method hasRegisteredApplication is never overridden by their sub classes, and could be made static. From my point of view, It should be a method of the class `ExternalFileType` (without the Optional parameter of course). 

The other concrete method is also never overridden and has only two very simple lines of code. It is only required in two places in the JabRefDesktop class which are only a few lines apart in the same method. Hence, I would move this method to the calling side, as it does not really help. 

With this said, I would convert the abstract class back to an interface without default methods which can be handled more easily (favor composition over inheritance).